### ISSUE TITLE
The memory usage is a little high 

### ISSUE THREAD

**Issue by xl95988** [qicosmos/ormpp#91]
![G88L(5E{JY5YI}$}1NLWUM6](https://user-images.githubusercontent.com/19475140/233609777-1c09b36e-c979-4ebc-a26b-ad3489b7939a.jpg)
![03EXKNH1U( LZL~`})EOXWL](https://user-images.githubusercontent.com/19475140/233609811-347f9064-cea6-4280-a22a-b7834bf48773.jpg)

**Comment by qicosmos** [comment]
死循环创建连接，会导致创建非常多的连接，因为内部发现如果连接池没有连接了，就会创建新的连接
https://github.com/qicosmos/ormpp/blob/master/include/connection_pool.hpp#L47

**Comment by xl95988** [comment]
I don't agree with you.  I set four connections when initializing the connection pool,  so 'https://github.com/qicosmos/ormpp/blob/master/include/connection_pool.hpp#L47' is never execute.
